### PR TITLE
[ARM] `az bicep install`: Support additional process architectures with Bicep CLI

### LINKED ISSUE #27333: Add newer process architecture support for Bicep installation
**Related command**
```sh
az bicep install
```

**Is your feature request related to a problem? Please describe.**
AzCLI's built-in Bicep support is limited to the following architectures:
- bicep-win-x64
- bicep-linux-x64
- bicep-linux-musl-x64
- bicep-osx-x64

However we also publish the following binaries:
- bicep-win-arm64
- bicep-linux-arm64
- bicep-osx-arm64

See here for the code which is figuring out the correct binary to install: https://github.com/Azure/azure-cli/blob/090024ddd24311b8daafe5d831cf5fe11cf7743c/src/azure-cli/azure/cli/command_modules/resource/_bicep.py#L271-L287

I suspect this is the cause behind https://github.com/Azure/bicep/issues/11734.

**Describe the solution you'd like**
AzCLI should be able to detect arm64 processors and install arm64 Bicep binaries

**Possible solution**
AzCLI has similar logic for helm - we could probably reuse some of this logic for architecture detection: https://github.com/Azure/azure-cli/blob/090024ddd24311b8daafe5d831cf5fe11cf7743c/src/azure-cli/azure/cli/command_modules/acr/helm.py#L338-L362

### PR DESCRIPTION
**Pull request by anthony-c-martin**
**Related command**
`az bicep`

**Description**
Support additional process architectures with Bicep CLI. Closes #27333.

**Testing Guide**
`az bicep install`

**History Notes**

[ARM] `az bicep install`: Support additional process architectures with Bicep CLI

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

**Comment by anthony-c-martin** [comment]
I didn't intend to add the "breaking change" title - must have copy+pasted that from another PR.

**Review by zhoxing-ms — DISMISSED** [review]

**Comment by zhoxing-ms** [comment]
@shenglol Could you please help review this Bicep related PR?

File: src/azure-cli/azure/cli/command_modules/resource/_bicep.py [hunk 1]
```diff
@@ -268,23 +276,37 @@ def _get_bicep_installed_version(bicep_executable_path):
     return _extract_version(installed_version_output)
 
 
-def _get_bicep_download_url(system, release_tag, target_platform=None):
-    download_url = f"https://downloads.bicep.azure.com/{release_tag}/{{}}"
-
-    if target_platform:
-        executable_name = "bicep-win-x64.exe" if target_platform == "win-x64" else f"bicep-{target_platform}"
-        return download_url.format(executable_name)
+def _has_musl_binaries():
```
Comment (shenglol) [inline src/azure-cli/azure/cli/command_modules/resource/_bicep.py:279]: nit: Might make sense to use the name `_has_musl_library_only` to indicate that this function checks for the presence of the musl library while explicitly excluding x86_64.

**Review by shenglol — APPROVED** [review]
Just added a minor comment. Otherwise looks good!

File: src/azure-cli/azure/cli/command_modules/resource/_bicep.py (hunk 1 above)
Comment (anthony-c-martin) [reply src/azure-cli/azure/cli/command_modules/resource/_bicep.py]: Fixed.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**